bundles/org.openhab.binding.unifiprotect/src/main/java/org/openhab/binding/unifiprotect/internal/handler/UnifiProtectCameraHandler.java

```diff
@@ -0,0 +1,684 @@
+        if (enableWebRTC) {
+            try {
+                rtsps = api.createRtspsStream(deviceId,
+                        List.of(ChannelQuality.HIGH, ChannelQuality.MEDIUM, ChannelQuality.LOW));
+            } catch (IOException e) {
+                logger.debug("Failed to create RTSP streams", e);
+            }
+        }
+        // update existing channels
+        updateStringChannel(UnifiProtectBindingConstants.CHANNEL_RTSP_URL_HIGH, rtsps != null ? rtsps.high : null);
+        updateStringChannel(UnifiProtectBindingConstants.CHANNEL_RTSP_URL_MEDIUM, rtsps != null ? rtsps.medium : null);
+        updateStringChannel(UnifiProtectBindingConstants.CHANNEL_RTSP_URL_LOW, rtsps != null ? rtsps.low : null);
+        updateStringChannel(UnifiProtectBindingConstants.CHANNEL_RTSP_URL_PACKAGE,
+                rtsps != null ? rtsps.packageUrl : null);
+
+        // Register new streams if available
+        if (enableWebRTC && rtsps != null) {
+            URI bc = null;
+            try {
+                TalkbackSession talkback = startTalkback();
+                bc = URI.create(talkback.url);
+            } catch (IOException e) {
+                logger.debug("Talkback not stupported: {}", e.getMessage());
+            }
+            final URI backChannel = bc;
+            // streamId -> sources (backchannel + main stream)
+            Map<String, List<URI>> streams = new LinkedHashMap<>();
+            BiConsumer<String, @Nullable String> add = (type, url) -> {
+                if (url != null && !url.isBlank()) {
+                    List<URI> sources = new ArrayList<>();
+                    sources.add(URI.create(url));
+                    if (backChannel != null) {
+                        sources.add(backChannel);
+                    }
+                    String fullStreamId = baseSourceId + ":" + type;
+                    streams.put(fullStreamId, sources);
+                    String webRTCId = UnifiProtectBindingConstants.CHANNEL_WEBRTC_URL + "-" + type;
+                    String webRTCUrl = media.getPlayBasePath() + "/" + fullStreamId;
+                    // set a property as well as a channel so users don't need to map to an Item
+                    getThing().setProperty(webRTCId, webRTCUrl);
+                    updateStringChannel(webRTCId, webRTCUrl);
+                }
+            };
+            add.accept("high", rtsps.high);
+            add.accept("medium", rtsps.medium);
+            add.accept("low", rtsps.low);
+            add.accept("package", rtsps.packageUrl);
+            media.registerHandler(this, streams);
+        } else {
+            media.registerHandler(this, Map.of());
+        }
+    }
+
+    private void addRemoveChannels(Camera camera) {
+        List<Channel> channelRemove = new ArrayList<>();
+        for (Channel existing : thing.getChannels()) {
+            channelRemove.add(existing);
+        }
+        updateThing(editThing().withoutChannels(channelRemove).build());
```

Comment:
I would opt to make this a little smarter and not remove everything and then add channels.

Maybe better to calculate the delta and only add/remove the differences.